src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -803,13 +817,12 @@ async_manifest_view::time_based_retention(
 
                 if (!eof) {
                     auto r = co_await cursor->next();
-                    if (
-                      r.has_failure()
-                      && r.error() == error_outcome::out_of_range) {
+                    if (r.has_value() && r.value() == false) {
```

Comment:
Why would it be corrupted? `false` just means EOF here and it's returned when `next` is called and the cursor is on the last spillover manifest.

Reply:
The code is correct but I think it's still error prone. I'd prefer to adjust truncation point segment by segment. But this can be changed later. The truncation of the whole archive is extreme so it's better to have an audit trails and to get to that state incrementally.